cvrf2csaf/section_handlers/product_tree.py

```diff
@@ -0,0 +1,129 @@
+import logging
+from ..common.common import SectionHandler
+
+
+class ProductTree(SectionHandler):
+    """ Responsible for converting the ProductTree section """
+
+    def __init__(self):
+        super().__init__()
+
+    def _process_mandatory_elements(self, root_element):
+        """ There are no mandatory elements in the ProductTree section """
+        pass
+
+    def _process_optional_elements(self, root_element):
+        self._handle_full_product_names(root_element)
+        self._handle_relationships(root_element)
+        self._handle_product_groups(root_element)
+
+        branches = self._handle_branches_recursive(root_element)
+        if branches is not None:
+            self.csaf['branches'] = branches
+
+    def _handle_full_product_names(self, root_element):
+        if not hasattr(root_element, 'FullProductName'):
+            return
+
+        full_product_names = []
+        for full_product_name in root_element.FullProductName:
+            fpn_to_add = {
+                'product_id': full_product_name.attrib['ProductID'],
+                'name': full_product_name.text,
+            }
+            if full_product_name.attrib.get('CPE'):
+                fpn_to_add['product_identification_helper'] = {'cpe': full_product_name.attrib['CPE']}
```

Comment:
It looks like that part of code occurs multiple times - as the handling is always the same for `full_product_name`. Maybe a function could help to prevent multiple copies?

Reply:
Sure, good one :) Implemented